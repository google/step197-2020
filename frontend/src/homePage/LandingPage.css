.landingPage {
    height:60vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.landingPage h1 {
    font-size: 64px;;
}