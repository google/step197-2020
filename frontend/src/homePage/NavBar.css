/* Add a black background color to the top navigation */
.navBar {
    background-color: #5A90A1;
    overflow: hidden;
    display:flex;
    justify-content:flex-end;
    padding:10px 0;
    position: fixed;
    top:0;
    width:100%;
  }