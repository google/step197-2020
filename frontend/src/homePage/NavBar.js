import React, {Component} from "react";
import css from "./NavBar.css";
import LoginButton from "./LoginButton.js";
import {motion} from "framer-motion"

class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.html = null;
    this.message = "Login";
    // Check if the user is logged in to display folders
    if (this.props.loginStatus) {
      this.html = (
        <LoginButton
          status="My Folders"
          color="#F4B400"
          fontSize="24px"
          clickFunction={this.props.folderClick}
        ></LoginButton>
      );
      this.message = "Logout";
    }
  }

  render() {
    return (
      <div className="navBar">
        <motion.div whileHover={{ scale: 1.1 }}>{this.html}</motion.div>
        <motion.div whileHover={{ scale: 1.1 }}>
          <LoginButton
            status={this.message}
            clickFunction={this.props.loginClick}
            color="#C4C4C4"
            fontSize="24px"
          ></LoginButton>
        </motion.div>
      </div>
    );
  }
}

export default NavBar;