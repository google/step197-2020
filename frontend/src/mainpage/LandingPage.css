.LandingPage{
    height:80vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.LandingPage h1{
    font-size: 64px;;
}