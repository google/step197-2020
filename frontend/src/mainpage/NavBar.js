import React from 'react';
//import './NavBar.css';
import LoginButton from './LoginButton.js';

class NavBar extends React.Component {
    constructor(props) {
      super(props);
    }
  
    render() {
      return (
        <div className="NavBar">
          <div className="Logo"></div>
          <LoginButton status="Login" color="#C4C4C4" fontSize="24px"></LoginButton>
        </div>
      );
    }
  }

export default NavBar;

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: https://bit.ly/CRA-PWA
//serviceWorker.unregister();