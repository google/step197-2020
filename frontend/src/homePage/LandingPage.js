import React from "react";
import "./LandingPage.css";
import LoginButton from "./LoginButton.js";
import { motion } from "framer-motion";

class LandingPage extends React.Component {
  constructor(props) {
    super(props);
    this.message = "Get Started";
    this.color = "#5A90A1";
    this.clickHandling = this.props.loginClick;
    if (this.props.loginStatus) {
      this.message = "My Folders";
      this.color = "#F4B400";
      this.clickHandling = this.props.folderClick;
    }
  }

  render() {
    return (
      <div className='landingPage'>
        <h1>Frame.Cards learning made easy</h1>
        <motion.div whileHover={{ scale: 1.2 }}>
          <LoginButton
            status={this.message}
            color={this.color}
            fontSize='32px'
            clickFunction={this.clickHandling}></LoginButton>
        </motion.div>
      </div>
    );
  }
}

export default LandingPage;