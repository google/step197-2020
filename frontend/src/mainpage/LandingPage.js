import React from 'react';
import './LandingPage.css';
import LoginButton from './LoginButton.js';

class LandingPage extends React.Component {
    constructor(props) {
      super(props);
    }
  
    render() {
      return (
        <div className="LandingPage">
          <h1>Frame.cards learning made easy</h1>
          <LoginButton status="Get Started" color="#5A90A1" fontSize="32px"></LoginButton>
        </div>
      );
    }
  }

export default LandingPage;