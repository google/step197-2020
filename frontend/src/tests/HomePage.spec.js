import React from "react";
import renderer from "react-test-renderer";
import LandingPage from "../homePage/LandingPage";
import NavBar from "../homePage/NavBar";

describe("Home Page component testing", () => {
  test("Landing page loads button with alternating text", () => {
    const component = renderer.create(<LandingPage loginStatus={true} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Nav Bar generates new tab when logged in", () => {
    const component = renderer.create(<NavBar loginStatus={true} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Nav Bar does not generate my folders tab when logged out", () => {
    const component = renderer.create(<NavBar loginStatus={false} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
