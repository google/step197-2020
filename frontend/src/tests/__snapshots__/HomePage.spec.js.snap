// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page component testing Landing page loads button with alternating text 1`] = `
<div
  className="LandingPage"
>
  <h1>
    Frame.cards learning made easy
  </h1>
  <div
    style={Object {}}
  >
    <div
      className="buttonBox"
    >
      <button
        className="MyButton"
        style={
          Object {
            "backgroundColor": "#F4B400",
            "fontSize": "32px",
          }
        }
      >
         
        My Folders
         
      </button>
    </div>
  </div>
</div>
`;

exports[`Home Page component testing Nav Bar does not generate my folders tab when logged out 1`] = `
<div
  className="NavBar"
>
  <div
    style={Object {}}
  />
  <div
    style={Object {}}
  >
    <div
      className="buttonBox"
    >
      <button
        className="MyButton"
        style={
          Object {
            "backgroundColor": "#C4C4C4",
            "fontSize": "24px",
          }
        }
      >
         
        Login
         
      </button>
    </div>
  </div>
</div>
`;

exports[`Home Page component testing Nav Bar generates new tab when logged in 1`] = `
<div
  className="NavBar"
>
  <div
    style={Object {}}
  >
    <div
      className="buttonBox"
    >
      <button
        className="MyButton"
        style={
          Object {
            "backgroundColor": "#F4B400",
            "fontSize": "24px",
          }
        }
      >
         
        My Folders
         
      </button>
    </div>
  </div>
  <div
    style={Object {}}
  >
    <div
      className="buttonBox"
    >
      <button
        className="MyButton"
        style={
          Object {
            "backgroundColor": "#C4C4C4",
            "fontSize": "24px",
          }
        }
      >
         
        Logout
         
      </button>
    </div>
  </div>
</div>
`;
